stdlib/src/builtin/simd.mojo

```diff
@@ -2275,39 +2276,87 @@ struct SIMD[type: DType, size: Int = simdwidthof[type]()](
         return self.reduce[mul_reduce_body, size_out]()
 
     @always_inline
-    fn reduce_and(self) -> Bool:
-        """Reduces the boolean vector using the `and` operator.
+    fn reduce_and[size_out: Int = 1](self) -> SIMD[type, size_out]:
+        """Reduces the vector using the bitwise and `&` operator.
+
+        Parameters:
+            size_out: The width of the reduction.
 
         Constraints:
-            The element type of the vector must be boolean.
+            `size_out` must not exceed width of the vector.
```

Comment:
Maybe consider adding a `constrained` for this?

Reply:
Its constrained already by calling reduce

Reply:
we should ask about having constraints as close to the caller as possible